app/questionnaire/navigation.py

```diff
@@ -34,147 +42,131 @@ def build_navigation(self, current_group_id, current_group_instance):
         if navigation_block is None or navigation_block.get('visible', True) is False:
             return None
 
-        summary_or_confirmation_blocks = self.schema.get_summary_and_confirmation_blocks()
-
         navigation = []
 
         for section in self.schema.sections:
             non_skipped_groups = self._get_non_skipped_groups(section)
             if not non_skipped_groups:
```

Comment:
Why aren't we using the Completeness class for this method? At the moment Navigation still seems to be evaluating skip conditions when that has already been done by the Completeness class. Ideally completeness would be the one stop shop for defining what state a section, group or block is in.

Reply:
Have addressed this now.